pgrx-macros/src/lib.rs

```diff
@@ -148,6 +148,13 @@ pub fn initialize(_attr: TokenStream, item: TokenStream) -> TokenStream {
     item
 }
 
+/// Declare a function as `#[pg_cast]` to indicate that it represents a Postgres cast
+/// `cargo pgrx schema` will automatically generate the underlying SQL
+#[proc_macro_attribute]
+pub fn pg_cast(_attr: TokenStream, item: TokenStream) -> TokenStream {
+    item
+}
```

Comment:
This should, if we go with `pg_cast` requiring `pg_extern`, specify that it requires `pg_extern`.

In either case, it should also document the valid arguments to the attribute (implicit, assignment), because these things are not made obvious by tooling, unlike how they would be in e.g. a function (which would emit its arguments and types into rustdoc, at least).

Not all `proc_macro_attribute`s in `pgrx-macros` do this, so you may be short on examples to imitate. It just has to cover the very basics, here.

Reply:
Went with a different approach of having `pg_cast` do the compile time validation and only after validation delegate to `PgExtern` for code-gen for the actual pg functions and casts. This avoids all the impossible states that the previous approach has.

Let me know what you think!